### ISSUE TITLE
[Question] What is the definition of GenerationID

### ISSUE THREAD

**Issue by chientranthien** [IBM/sarama#2775]
There is a function of `ConsumerGroupSession` called `GenerationID`

- What is the definition of `GenerationID`? 
- Is it true that `GenerationID` of a consumer group always increases each consumer-group rebalancing?

**Comment by dnwe** [comment]
@chientranthien yes, this is sort of exposing the internals of Kafka, but the ConsumerGroup has an ID integer that increases whenever a rebalance or change of membership occurs in the group, so that the server-side group coordinator can be sure that all members of the group are aware of the configuration/memberships and it can ignore any requests that come in referring to an old/outdated generation, forcing consumers to correctly re-join the group at the new generation

https://github.com/apache/kafka/blob/60e845626d8a465a8cfe68bb2d7d4b88d622634e/core/src/main/scala/kafka/coordinator/group/GroupMetadata.scala#L583-L598

You might find https://chrzaszcz.dev/2019/06/kafka-rebalancing/ some useful background reading as well
